### PR TITLE
Asset: Separate preparation of creation of VC Asset from dispatch

### LINKED ISSUE #213: [C4GT] Asset: Separate preparation of creation of VC Asset from dispatch
## Description
Separate preparation of VC Asset Entry and dispatch of extrinsic to chain. This would help when bulk transaction takes place so we just prepare all the required extrinsics and then execute these transaction in batches by dumping them into a promise using 
a batch sign & submit utility ex, `authorizeBatch` which can take array of extrinsics, instead of just one.

## Goals

- [ ] Creation of new export function named `prepareCreateVcExtrinsic()`, which will contain creation of asset entry and `Did.AuthorizeTx()`. 
- [ ] Integration of above method into existing `dispatchCreateVcToChain()`

## Implementation Details
Developers/Contributors can refer to example PRs: #188, #182

---- 

## Product Name
CORD.JS

## Organization Name
Dhiway

## Domain
Blockchain, SDK Development, API

## Tech Skills Needed
Typescript 

## Mentor(s)
@amarts @vatsa287

## Complexity
[Low]

## Category
[Performance Improvement], [Feature]

## Sub Category
 [API], [Refactoring], [Backend], [Beginner friendly]

### NOTE: 
This is a `good-first-issue` meaning it is open for new contributors. If you have already worked on similar issues, please avoid getting assigned on these & start working on next level issues. Please read this [discussion](https://github.com/dhiway/cord/discussions/396).

### PR DESCRIPTION
**Pull request by masterismail**
fixes #213

### CODE REVIEW

**Comment by masterismail** [comment]
@vatsa287 @amarts please review.

File: packages/asset/src/Asset.chain.ts [hunk 1]
```diff
@@ -78,44 +78,66 @@ export async function dispatchCreateToChain(
...
+  authorizationUri: AuthorizationUri,
+  assetEntryUri: AssetUri,
+  signCallback: SignExtrinsicCallback
+): Promise<AssetUri> {
+  try {
+    const extrinsic = await prepareCreateVcExtrinsic(
+      assetQty,
+      digest,
+      creator,
+      authorizationUri,
```
Comment (vatsa287) [inline packages/asset/src/Asset.chain.ts:151]: `prepareVcExtrinsic` expects `authorAccount` as a parameter after `creator`, which is missing while calling the method.

**Review by vatsa287 — CHANGES_REQUESTED** [review]
Please keep a newline after every method end.

File: packages/asset/src/Asset.chain.ts (hunk 1 above)
Comment (masterismail) [reply packages/asset/src/Asset.chain.ts:151]: Done !

**Review by vatsa287 — APPROVED** [review]
LGTM

**Review by amarts — APPROVED** [review]
Nitpick: A good PR should implement only focused area, and fix whitespace issues only in the related area. For fixing any other lines, either send them as a separate PR or separate commit in the same PR (ie, one PR can have multiple commits) so for reviewer, it is easier to focus on the 'logic' changes.

That practise helps in getting PRs reviewed quickly.

**Comment by amarts** [comment]
Ah! anyways, I see some merge conflicts here, can you rebase the whole PR ?

**Comment by masterismail** [comment]
> Nitpick: A good PR should implement only focused area, and fix whitespace issues only in the related area. For fixing any other lines, either send them as a separate PR or separate commit in the same PR (ie, one PR can have multiple commits) so for reviewer, it is easier to focus on the 'logic' changes.
> 
> That practise helps in getting PRs reviewed quickly.

 Resolved the merged conflict, thank you for the feedback ! I'll keep PRs more focused in the future.

**Comment by VedantKhairnar** [comment]
@vatsa287 If the PR is good, can we merge it? Thanks

**Review by amarts — APPROVED** [review]

**Pull request merged**